SofaScoreAcademy/SofaScoreAcademy/Views/MatchViewCell.swift

```diff
@@ -0,0 +1,204 @@
+//
+
+import UIKit
+import SnapKit
+import SofaAcademic
+
+class MatchViewCell: UITableViewCell {
+    
+    enum WinnerCode {
+        case home
+        case away
+    }
+    
+    private let teamOneNameLabel = UILabel()
+    private let teamTwoNameLabel = UILabel()
+    private let teamOneScoreLabel = UILabel()
+    private let teamTwoScoreLabel = UILabel()
+    private let teamOneImageView = UIImageView()
+    private let teamTwoImageView = UIImageView()
+    private let matchTimeLabel = UILabel()
+    private let matchStatusLabel = UILabel()
+    private let matchCellVerticalDivider = UIView()
+    
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        addViews()
+        setupConstraints()
+        styleViews()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func configure(with event: Event) {
+        teamOneNameLabel.text = event.teamOneName
+        teamTwoNameLabel.text = event.teamTwoName
+        matchTimeLabel.text = event.formattedStartTime
+        let teamOneScore = event.teamOneScore ?? 0
+        let teamTwoScore = event.teamTwoScore ?? 0
+        teamOneScoreLabel.text = getTeamScoreDescription(teamOneScore, status: event.matchStatus)
+        teamTwoScoreLabel.text = getTeamScoreDescription(teamTwoScore, status: event.matchStatus)
+        updateStatus(for: event.matchStatus, with: event.matchTime, teamOneScore: teamOneScore, teamTwoScore: teamTwoScore)
+    }
+    
+    private func getTeamScoreDescription(_ score: Int, status: MatchStatus) -> String? {
+        if status == .isFinished || status == .inProgress {
+            return score.description
+        }
+        return nil
+    }
+
+    private func updateStatus(for status: MatchStatus, with time: Int?, teamOneScore: Int, teamTwoScore: Int) {
+        let winnerCode: WinnerCode? = {
+            if status == .isFinished {
+                if teamOneScore > teamTwoScore {
+                    return .home
+                } else if teamOneScore < teamTwoScore {
+                    return .away
+                }
+            }
+            return nil
+        }()
+        let teamColors = getTeamNameColors(for: status, winnerCode: winnerCode)
+        let scoreColors = getTeamScoreColors(for: status, winnerCode: winnerCode)
+        matchStatusLabel.text = status == .notStarted ? "-" : (status == .inProgress ? "\(time ?? 0)'" : "FT")
+        matchStatusLabel.textColor = status == .inProgress ? .specificLive : .onSurfaceOnSurfaceLv2
+        teamOneNameLabel.textColor = teamColors.left
+        teamTwoNameLabel.textColor = teamColors.right
+        teamOneScoreLabel.textColor = scoreColors.left
+        teamTwoScoreLabel.textColor = scoreColors.right
+    }
+    
+    func getTeamNameColors(for status: MatchStatus, winnerCode: WinnerCode?) -> ValuePair<UIColor> {
```

Comment:
mislim da i ovu i getTeamScoreColors mozes isto private stavit cak ih i mozes stavit u private extension ako ti se to svida